### PR TITLE
Seed propagation for injected productions

### PR DESCRIPTION
**Pull request by mpuccio**
Hi @benedikt-voelkel, @chiarazampolli, @sawenzel, @fgrosa,

how it was done before the seed was not propagated to the signal generator if a custom configuration of pythia8 was passed in the ini file. What I propose here is that we get a random number starting from the jobid using the python random number generator and then we increase it by 1 for each TF. The maximum seed is truncated to fit the Pythia8 requirements.
Then this number is given as general seed to o2-sim, that already gives it to Pythia and it will be the same we set from outside once AliceO2Group/AliceO2#8727 will be merged.

In the case of the HF injection we were also specifying the process in the bash script, thus leading to the on-the-fly generation of a pythia configuration that was overwriting the HF defaults (it was then hacked back in via the confKeyValues). Now this should be solved and I added the support for the SPLITID as seed in the HF bash script.

If you have any comments or feedback, it is very welcome!
Cheers,
Max

### CODE REVIEW

**Comment by fgrosa** [comment]
Hi @mpuccio thanks a lot for this implementation, it looks very good for me! 👍

File: MC/run/PWGHF/run_pp_HFtriggers_bbbar_userhook.sh [hunk 1]
```diff
@@ -11,24 +11,21 @@
...
 NSIGEVENTS=${NSIGEVENTS:-1}
 NBKGEVENTS=${NBKGEVENTS:-1}
 NTIMEFRAMES=${NTIMEFRAMES:-1}
-
+SPLITID=${SPLITID:-None}
 # create workflow
 
 #ccbar filter
-${O2DPG_ROOT}/MC/bin/o2dpg_sim_workflow.py -eCM 13500 -col pp -gen pythia8 -proc "inel" -j ${NWORKERS} -ns ${NSIGEVENTS} -tf ${NTIMEFRAMES} -interactionRate 500000 -confKey "Diamond.width[2]=6.;GeneratorPythia8.config=${O2DPG_ROOT}/MC/config/PWGHF/pythia8/generator/pythia8_charmtriggers_with_decays.cfg" -e TGeant4 -mod "--skipModules ZDC" \
+${O2DPG_ROOT}/MC/bin/o2dpg_sim_workflow.py -eCM 13500 -col pp -gen pythia8 -j ${NWORKERS} -ns ${NSIGEVENTS} -tf ${NTIMEFRAMES} -interactionRate 500000 -confKey "Diamond.width[2]=6.;" -e ${SIMENGINE} -seed ${SPLITID} -mod "--skipModules ZDC" \
```
Comment (benedikt-voelkel) [inline MC/run/PWGHF/run_pp_HFtriggers_bbbar_userhook.sh:26]: Would that be interpreted as a string by `argparse`? And does the seeding of python's `random` work with that string of `"None"`?
Same in `MC/run/PWGHF/run_pp_HFtriggers_ccbar_userhook.sh`

**Comment by benedikt-voelkel** [comment]
Thanks @mpuccio !
That looks to me as if it should work now. I just have a brief comment/question above.
Eventually, we need to find a way to make it generic as soon as we start to actually use generators other than pythia8. But we can put that on the agenda

File: MC/run/PWGHF/run_pp_HFtriggers_bbbar_userhook.sh (hunk 1 above)
Comment (mpuccio) [reply MC/run/PWGHF/run_pp_HFtriggers_bbbar_userhook.sh]: Very good point! I updated this part of the code to avoid passing the seed if SPLITID is not set

**Pull request merged**